data/json/npcs/TALK_CITY_COP.json

```diff
@@ -2,9 +2,20 @@
   {
     "type": "talk_topic",
     "id": "TALK_CITY_COP",
-    "dynamic_line": "STOP!  Put your hands in the air!  Ha, startled you, didn't I... There is no law anymore...",
+    "dynamic_line": {
+      "u_has_var": "talked_to_survivor_cop",
+      "type": "dialogue",
+      "context": "survivor_cop",
+      "value": "yes",
+      "no": "STOP, Put your hands in the air! Ha, startled you didn't I...there is no law anymore...",
+      "yes": "Hi there, <name_g>."
+    },
     "responses": [
-      { "text": "What are you doing here?", "topic": "TALK_CITY_COP_INTRO" },
+      {
+        "text": "What are you doing here?",
+        "topic": "TALK_CITY_COP_INTRO",
+        "effect": { "u_add_var": "talked_to_survivor_cop", "type": "dialogue", "context": "survivor_cop", "value": "yes" }
```

Comment:
You can do "u_add_var" as a speaker_effect now to guarantee that it will run immediately after the cop speaks the dynamic line for the first time.